borg/testsuite/repository.py

```diff
@@ -227,6 +228,28 @@ def test_ignores_commit_tag_in_data(self):
             io = self.repository.io
             assert not io.is_committed_segment(io.get_latest_segment())
 
+    def test_moved_deletes_are_tracked(self):
+        self.repository.put(H(1), b'wasd')
+        self.repository.put(H(2), b'hjkl')
+        self.repository.commit()
+        self.repository.delete(H(1))
+        self.repository.commit()
+        last_segment = self.repository.io.get_latest_segment()
+        num_deletes = 0
+        for tag, key, offset, data in self.repository.io.iter_objects(last_segment, include_data=True):
+            if tag == TAG_DELETE:
+                assert key == H(1)
+                num_deletes += 1
+        assert num_deletes == 1
+        assert last_segment in self.repository.compact
+        self.repository.put(H(3), '↑←↓→'.encode())
+        self.repository.commit()
+        assert last_segment not in self.repository.compact
+        assert not self.repository.io.segment_exists(last_segment)
+        last_segment = self.repository.io.get_latest_segment()
+        for tag, key, offset, data in self.repository.io.iter_objects(last_segment, include_data=True):
+            assert tag != TAG_DELETE
```

Comment:
note for merging into master: last_segment = self... latest_segment() - 1 because the extra small/fast commit segment will be the last one.